### ISSUE TITLE
gollum breaks with mustache-sinatra 1.0.0: undefined method 'view_namespace='

### ISSUE THREAD

**Issue by plujon** [gollum/gollum#1724]
Determining whether this is a bug in gollum, omnigollum, mustache-sinatra 1.0.0, or bundler is left as an exercise for the reader.  But the people who will hit this are gollum users, so I'm filing it here.

cat Gemfile:

    gem 'gollum'
    gem 'omnigollum'
    source 'https://rubygems.org'

bundle install
bundle exec rackup
chrome http://localhost:9292/

NoMethodError at /gollum/create/Home
undefined method `view_namespace=' for #<Class:0x00005593cff615e0>

The problem is caused by use of mustache-sinatra-1.0.0.

To avoid this problem, change the Gemfile:

    gem 'gollum'
    gem 'omnigollum'
    gem 'mustache-sinatra', '>= 1.0.1'
    source 'https://rubygems.org'

The same bug can be seen with the simpler (but perhaps, more confusing):

    gem 'gollum'
    gem 'mustache-sinatra', '1.0.0'
    source 'https://rubygems.org'

What is especially mind boggling about this is that mustache-sinatra has not changed in 6 years.  Why bundler includes it when omnigollum is present, I have no clue, but Gollum's dependency on mustache-sinatra should probably be changed to specify '>= 1.0.1', '< 2' to avoid the problem.

**Comment by benjaminwil** [comment]
Thank you for this detailed report.

Gollum's Gemspec should be installing `mustache-sinatra v1.0.1` currently.

The version constraint `~> 1.0` is equivalent to `'>= 1.0', '< 2'`.

To test, I created a new Gemfile:

```
gem 'gollum'
gem 'omnigollum'

source 'https://rubygems.org'
```

And I confirmed the issue you're seeing.

However, when I remove `omnigollum` and run `bundle update`, `mustache-sinatra` is updated to `1.0.1` like we want.

`bundle update` outputs:

```
...
Fetching mustache 0.99.8 (was 1.1.1)
...
Installing mustache 0.99.8 (was 1.1.1)
Fetching mustache-sinatra 1.0.1 (was 1.0.0)
Installing mustache-sinatra 1.0.1 (was 1.0.0)
...
Bundle updated!
```

Then, if I go back into the Gemfile, remove `gollum` entirely, and add `omnigollum`, we go back to the bad version of `mustache-sinatra`:

```
...
Using mustache-sinatra 1.0.0 (was 1.0.1)
...
Using mustache 1.1.1 (was 0.99.8)
...
Bundle updated!
```

So I think this issue is actually related to `omnigollum` and would only affect `omnigollum` users. Perhaps we should report this issue on that repo.

**Comment by dometto** [comment]
This is very odd. We depend on `mustache-sinatra ~> 1.0`, which depends on `mustache ~> 1.0`. Yet `bundle install` gives us `mustache v0.99.8`. 😕

**Comment by dometto** [comment]
Omnigollum simply depends on `mustache ~> 1.0`, so that should be perfectly compatible with our dependency. Not sure what's going on here. Is this a `bundle` fail?

**Comment by plujon** [comment]
Because mustache-sinatra 1.0.0 specified fewer version constraints in mustache-sinatra.gemspec than does mustache-sinatra 1.0.1, I think bundler is forced into a lose-lose: it has to choose between installing an older mustache-sinatra (1.0.0) and newer mustache (1.1.1), or a newer mustache-sinatra (1.0.1) with an older mustache (0.99.8).  Specifying the 1.0.1 version in the Gemspec helps bundler decide.  I think.

**Comment by benjaminwil** [comment]
I see no harm in explicitly requiring `>= 1.0.1`, `< 2` in the gemspec! It's just too bad ~ :joy:

**Comment by bartkamphorst** [comment]
@plujon Would you mind preparing a PR for this?

**Comment by dometto** [comment]
Fixed in #1749
